.github/workflows/deploy_main_image.yaml

```diff
@@ -0,0 +1,44 @@
+# This workflow creates docker image based on main branch and publishes it to project's container registry.
+# It doesn't use any cache and performs clean build, so it may take longer than other jobs.
+
+on: 
+  push:
+    branches:
+      - 'main'
+jobs:
+  deploy-main-image:
+    name: Build and deploy main docker image
+    runs-on: ubuntu-latest
+    env:
+      PLATFORM: linux/amd64,linux/arm64
+      MAIN_IMAGE: ghcr.io/${{ github.repository_owner }}/rover-soft:stable
+
+    steps:
+      - name: lowercase image name
+        run: |
+          echo "MAIN_IMAGE=${MAIN_IMAGE,,}" >>${GITHUB_ENV}
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and Push image
```

Comment:
Adding a newline here would look better

Reply:
Fixed in [efbebdd](https://github.com/SKA-Robotics/rover-soft/commit/efbebdd81c57cee360b328e366ee7f98abd9dc34)